collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -78,17 +77,24 @@ public void onReceive(Context context, Intent intent) {
 
     private boolean isFormAutoSendOptionEnabled(NetworkInfo currentNetworkInfo) {
         // make sure autosend is enabled on the given connected interface
-        String autosend = (String) GeneralSharedPreferences.getInstance().get(PreferenceKeys.KEY_AUTOSEND);
-        boolean sendwifi = autosend.equals("wifi_only");
-        boolean sendnetwork = autosend.equals("cellular_only");
-        if (autosend.equals("wifi_and_cellular")) {
-            sendwifi = true;
-            sendnetwork = true;
+        Set<String> multiAutoSend = (Set<String>) GeneralSharedPreferences.getInstance().get(PreferenceKeys.KEY_MULTI_AUTOSEND);
+        boolean sendWifi = false;
+        boolean sendNetwork = false;
+
+        for (String option : multiAutoSend) {
```

Comment:
No need to iterate here. `sendWifi = multiAutoSend.contains("wifi")` and `sendNetwork = multiAutoSend.contains("cellular")`.